packages/components/src/__future__/Tag/RemoveableTag.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { HTMLAttributes } from "react"
+import classNames from "classnames"
+import { OverrideClassName } from "~types/OverrideClassName"
+import { RemoveButton } from "./RemoveButton"
+import { Tag, TagProps } from "./Tag"
+import styles from "./RemoveableTag.module.scss"
+
+export type RemoveableTagProps = {
+  removeButtonProps: {
+    "aria-label": string
+    onClick: () => void
+  }
+} & TagProps &
+  OverrideClassName<HTMLAttributes<HTMLSpanElement>>
```

Comment:
We have indeed forgotten to make `Tag` extend HTML Attributes, so that's a good catch. Extend the attributes over there instead, and have this component extend Tag.

```suggestion
} & OverrideClassName<TagProps>
```